File: SampleBundle/RadialPosition.vb

Public Class RadialPosition

    Public Function GetRadialPosition(ItemNumber As Byte, ItemCount As Byte, canvasheight As Integer) As (x1 As Integer, y1 As Integer)
        '// Determine position of the button on the dial radius
        Dim ang As Double = (ItemNumber * ((2 * Math.PI) / ItemCount))
        Dim rad As Integer = canvasheight / 2
        Dim x As Integer = ((Math.Cos(ang) * rad) + rad)
        Dim y As Integer = (rad - (Math.Sin(ang) * rad))
        Return (x, y)
    End Function

End Class

File: SampleCSharp/RadialPosition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SampleCSharp
{
    class RadialPosition
    {
        public Tuple<int, int> GetPosition(byte ItemNumber, byte ItemCount, int CanvasHeight)
        {
            // Determine position of the button on the dial radius
            double ang = (ItemNumber * ((2 * Math.PI) / ItemCount));
            int rad = CanvasHeight / 2;
            int x = Convert.ToInt16(((Math.Cos(ang) * rad) + rad));
            int y = Convert.ToInt16((rad - (Math.Sin(ang) * rad)));
            return Tuple.Create (x,y);
        }
    }
}
